src/mpz.rs

```diff
@@ -168,36 +191,37 @@ impl Mpz {
             __gmpz_get_str(vector.as_mut_ptr() as *mut _, base as c_int, &self.mpz);
 
             let mut first_nul = None;
-            let mut index : usize = 0;
-            for elem in &vector {
+            for (index, elem) in vector.iter().enumerate() {
                 if *elem == 0 {
                     first_nul = Some(index);
                     break;
                 }
-                index += 1;
             }
```

Comment:
Could be simplified to
```rust
let first_nul = vector.iter().position(|elem| *elem == 0);
```

Reply:
Added the `.unwrap_or` onto it as well :)